examples/bubble_example.py

```diff
@@ -0,0 +1,343 @@
+        speed_limit = int(obs.waypoint_paths[0][0].speed_limit)
+        return (
+            random.randint(speed_limit - 15, speed_limit + 4),
+            random.randint(-1, 1),
+        )
+
+
+def register_dummy_locator(interface, name="dummy_agent-v0"):
+    class DummyAgent(Agent):
+        """This is just a place holder that is used for the default agent used by the bubble."""
+
+        def act(self, obs: Observation) -> Tuple[float, float]:
+            acceleration = 0.0
+            angular_velocity = 0.0
+            return (acceleration, angular_velocity)
+
+    # Register the dummy agent for use by the bubbles
+    #  referenced as `"<module>:<locator>"` (i.e. `"examples:dummy_agent-v0"`)
+    register(
+        name,
+        entry_point=lambda **kwargs: AgentSpec(
+            interface=interface,
+            agent_builder=DummyAgent,
+        ),
+    )
+
+
+def create_moving_bubble(
+    follow_vehicle_id=None, follow_agent_id=None, social_agent_name="dummy_agent-v0"
+):
+    assert follow_vehicle_id or follow_agent_id, "Must follow a vehicle or agent"
+    assert not (
+        follow_vehicle_id and follow_agent_id
+    ), "Must follow only one of vehicle or agent"
+
+    follow = dict()
+    exclusion_prefixes = ()
+    if follow_vehicle_id:
+        follow = dict(follow_vehicle_id=follow_vehicle_id)
+        exclusion_prefixes = (follow_vehicle_id,)
+    else:
+        follow = dict(follow_actor_id=follow_agent_id)
+
+    bubble = Bubble(
+        zone=PositionalZone(pos=(0, 0), size=(20, 40)),
+        actor=SocialAgentActor(
+            name="dummy0", agent_locator=f"examples:{social_agent_name}"
+        ),
+        exclusion_prefixes=exclusion_prefixes,
+        follow_offset=(0, 0),
+        margin=5,
+        **follow,
+    )
+    return bubble
+
+
+def resolve_agent_missions(
+    scenario: Scenario,
+    start_time: float,
+    run_time: float,
+    count: int,
+):
+    # pytype: disable=attribute-error
+    agent_missions = {
+        f"agent-{mission.vehicle_spec.veh_id}": mission
+        for mission in random.sample(
+            scenario.history_missions_for_window(
+                start_time, start_time + run_time, run_time / 2
+            ),
+            k=count,
+        )
+    }
+    # pytype: enable=attribute-error
+
+    return agent_missions
+
+
+def main(
+    script: str,
+    example: str,
+    headless: bool,
+    seed: int,
+    episodes: int,
+    start_time: float,
+    run_time: float,
+    num_agents: float,
+):
+    logger = logging.getLogger(script)
+    logger.setLevel(logging.INFO)
+
+    logger.debug("initializing SMARTS")
+
+    timestep = 0.1
+    run_steps = int(run_time / timestep)
+    social_interface = AgentInterface.from_type(
+        AgentType.Direct,
+        done_criteria=DoneCriteria(
+            not_moving=False,
+            off_road=False,
+            off_route=False,
+            on_shoulder=False,
+            wrong_way=False,
+        ),
```

Comment:
Maybe we could add a `ColiisionOnlyDoneCriteria` to `agent_interface.py` as a short-cut for this, as this may be a common use case?